conans/client/configure_environment.py

```diff
@@ -98,8 +98,8 @@ def _gcc_env(self):
 
     @property
     def command_line_env(self):
-        if self.os == "Linux" or self.os == "Macos" or self.os == "FreeBSD":
-            if self.compiler == "gcc" or "clang" in str(self.compiler):
+        if self.os == "Linux" or self.os == "Macos" or self.os == "FreeBSD" or self.os == "SunOS":
+            if self.compiler == "gcc" or "clang" in str(self.compiler) or "cc" in str(self.compiler):
```

Comment:
`"cc" ==  self.compiler` better than `"cc" in str(self.compiler)`
Could be error prone.
But I agree with @memsharded calling it sun-cc or something like that.